client/layout/logged-out-design.jsx

```diff
@@ -11,21 +13,35 @@ import React from 'react';
 import MasterbarMinimal from 'layout/masterbar/minimal';
 import ThemesHead from 'my-sites/themes/head';
 
-const LayoutLoggedOutDesign = ( { tier = 'all' } ) => (
-	<div className="wp is-section-design has-no-sidebar">
-		<ThemesHead tier={ tier } />
-		<MasterbarMinimal url="/" />
-		<div id="content" className="wp-content">
-			<div id="primary" className="wp-primary wp-section" />
-			<div id="secondary" className="wp-secondary" />
+const LayoutLoggedOutDesign = ( { section, hasSidebar, tier = 'all' } ) => {
+	const sectionClass = section ? ' is-section-' + section : '';
+	const classes = classNames( 'wp', sectionClass, {
+		'has-no-sidebar': ! hasSidebar
+	} );
+
+	return (
+		<div className={ classes }>
+			<ThemesHead tier={ tier } />
+			<MasterbarMinimal url="/" />
+			<div id="content" className="wp-content">
+				<div id="primary" className="wp-primary wp-section" />
+				<div id="secondary" className="wp-secondary" />
+			</div>
+			<div id="tertiary" className="wp-overlay fade-background" />
 		</div>
-		<div id="tertiary" className="wp-overlay fade-background" />
-	</div>
-)
+	);
+}
 
 LayoutLoggedOutDesign.displayName = 'LayoutLoggedOutDesign';
 LayoutLoggedOutDesign.propTypes = {
+	section: React.PropTypes.string,
+	hasSidebar: React.PropTypes.bool,
 	tier: React.PropTypes.string
 }
 
-export default LayoutLoggedOutDesign;
+export default connect(
+	( state ) => {
```

Comment:
We're losing props (i.e. `tier`) here. This needs to be `( state, props ) => Object.assign( {}, props, { ... } )` (cf the `ThemesListFetcher` data component).